File: python/sum_of_number_and_reverse.py

#        Python
# Time complexity -  O (NlogN)
# Space complexity - O (1)

class Solution(object):

    def sumOfNumberAndReverse(self, num: int) -> bool:
        for x in range(num + 1):
            if x + self.reverse_number(x) == num:
                return True
        return False

    def reverse_number(self, n: int) -> int:
        reversed_n = 0
        while n > 0:
            reversed_n = reversed_n * 10 + n % 10
            n //= 10
        return reversed_n

File: go/sum_of_number_and_reverse.go
package main

//        Go
// Time complexity -  O (NlogN)
// Space complexity - O (1)

func reverseNumber(n int) int {
	reversed := 0
	for n > 0 {
		reversed = reversed*10 + n%10
		n /= 10
	}
	return reversed
}

func sumOfNumberAndReverse(num int) bool {
	for x := 0; x <= num; x++ {
		if x+reverseNumber(x) == num {
			return true
		}
	}
	return false
}
